contractcourt/chain_arbitrator.go

```diff
@@ -340,18 +339,17 @@ func (c *ChainArbitrator) resolveContract(chanPoint wire.OutPoint,
 		return err
 	}
 
-	if arbLog != nil {
-		// Once this has been marked as resolved, we'll wipe the log
-		// that the channel arbitrator was using to store its
-		// persistent state. We do this after marking the channel
-		// resolved, as otherwise, the arbitrator would be re-created,
-		// and think it was starting from the default state.
-		if err := arbLog.WipeHistory(); err != nil {
-			return err
-		}
-	}
-
+	// Now that the channel has been marked as fully closed, we'll stop
+	// both the channel arbitrator and chain watcher for this channel if
+	// they're still active.
+	var arbLog ArbitratorLog
```

Comment:
It seems that locking is unnecessary for chainArb.Stop() and chainWatcher.Stop(), and should only be used while removing the chainArb and chainWatcher from their respective maps. (Please discard my comment if this is untrue and the whole process needs locking.).

Reply:
Updated to only hold the lock while reading from the map. All other operations directly on the objects are goroutine safe.